azext_edge/edge/util/file_operations.py

```diff
@@ -0,0 +1,109 @@
+# ----------------------------------------------------------------------------------------------
+
+import csv
+import json
+import yaml
+import os
+from pathlib import PurePath
+from typing import List, Optional
+from azure.cli.core.azclierror import FileOperationError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+def convert_file_content_to_json(file_path: str):
+    extension = file_path.split(".")[-1]
+    content = read_file_content(file_path)
+    if extension == "json":
+        try:
+            return json.loads(content)
+        except json.JSONDecodeError as e:
+            raise FileOperationError(f"Invalid JSON syntax: {e}")
+    elif extension == "yaml":
+        try:
+            return yaml.safe_load(content)
+        except yaml.YAMLError as e:
+            raise FileOperationError(f"Invalid YAML syntax: {e}")
+    elif extension == "csv":
+        try:
+            with open(file_path, "r", encoding="utf-8") as f:
+                result = list(csv.DictReader(f))
+            return result
+        except csv.Error as e:
+            raise FileOperationError(f"Invalid CSV syntax: {e}")
+    raise FileOperationError(f"Detected {extension} extension is not supported.")
+
+
+def dump_content_to_file(
+    content: List[dict],
+    file_name: str,
+    extension: str,
+    fieldnames: Optional[List[str]] = None,
+    output_dir: Optional[str] = None,
+    replace: bool = False
+) -> PurePath:
+    output_dir = normalize_dir(output_dir)
+    file_path = os.path.join(output_dir, f"{file_name}.{extension}")
+    if os.path.exists(file_path):
+        if not replace:
+            raise FileExistsError(f"File {file_path} already exists. Please choose another file name or add replace.")
+        logger.warning(f"The file {file_path} will be overwritten.")
+    if extension.endswith("csv"):
+        with open(file_path, "w", newline="", encoding="utf-8") as f:
+            if not fieldnames:
+                fieldnames = content[0].keys()
+            writer = csv.DictWriter(f, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerows(content)
+        return file_path
+
+    # These let you dump to a string before writing to file
+    if extension == "json":
+        content = json.dumps(content, indent=2)
+    elif extension == "yaml":
+        content = yaml.dump(content)
+    with open(file_path, "w", encoding="utf-8") as f:
+        f.write(content)
+
+    return file_path
+
+
+def normalize_dir(dir_path: Optional[str] = None) -> PurePath:
+    if not dir_path:
+        dir_path = "."
+    if "~" in dir_path:
+        dir_path = os.path.expanduser(dir_path)
+    dir_path = os.path.abspath(dir_path)
+    dir_pure_path = PurePath(dir_path)
+    if not os.path.exists(str(dir_pure_path)):
+        os.makedirs(dir_pure_path, exist_ok=True)
+
+    return dir_pure_path
+
+
+def read_file_content(file_path: str, read_as_binary: bool = False):
```

Comment:
A new version of this function already exists in common

Reply:
dammit